js/src/locale.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Simple module for localization of strings.
+ *
+ * How to use:
+ * 1. Create localization file with following JSON format:
+ * {
+ *   "en" :{
+ *     "how": "How do you want your egg today?",
+ *     "boiledEgg": "Boiled egg",
+ *     "softBoiledEgg": "Soft-boiled egg",
+ *     "choice": "How to choose the egg",
+ *     "buyQuestion": "I'd like to buy {0} eggs, {1} dollars each."
+ *   },
+ *
+ *   "it": {
+ *     "how": "Come vuoi il tuo uovo oggi?",
+ *     "boiledEgg": "Uovo sodo",
+ *     "softBoiledEgg": "Uovo alla coque",
+ *     "choice": "Come scegliere l'uovo",
+ *     "buyQuestion": "Mi piacerebbe comprare {0} uova, {1} dollari ciascuna."
+ *   }
+ * }
+ * 
+ * 2. Use Locale's module load method, specifying path to your localization file and set chosen language, e.g.:
+ *    > Locale.load('static/locale/locale.json');
+ *    > Locale.setLanguage('en');
+ * 
+ * 3. Use localization strings:
+ *    > console.log(Locale.tr.how);
+ *    How do you want your egg today?
+ *    > console.log(Locale.tr.boiledEgg);
+ *    Boiled egg
+ * 
+ * 4. String formatting
+ *    > console.log(Locale.tr.buyQuestion.format(10, 0.5));
+ *    I'd like to buy 10 eggs, 0.5 dollars each.
+ */
+
+export class Locale {
+    /**
+     * Loads translation object from external json file.
+     *  
+     * @param {String} path Path to resource.
+     * @return {Promise}
+     */
+    static async load(path) {
+        if (!this.__enabled) {
+            Promise.resolve()
+        }
+
+        await fetch(path)
+        .then((resp) => resp.json())
+        .then((resp) => {
+            this.loadFromObject(resp);
+        });
+    }
+
+    /**
+     * Sets language used by module.
+     *
+     * @param {String} lang 
+     */
+    static setLanguage(lang) {
+        this.__enabled = true;
+        this.language = lang;
+    }
+
+    /**
+     * Returns reference to translation object for current language.
+     *
+     * @return {Object}
+     */
+    static get tr() {
+        return this.__trObj[this.language];
+    }
+
+    /**
+     * Loads translation object from existing object (binds existing object).
+     *
+     * @param {Object} trObj 
+     */
+    static loadFromObject(trObj) {
+        this.__trObj = trObj;
+        for (const lang of Object.values(this.__trObj)) {
+            for (const str of Object.keys(lang)) {
+                lang[str] = new LocalizedString(lang[str]);
+            }
+        }
+    }
+}
+
+Locale.__enabled = false;
+
+/**
+ * Extended string class used for localization.
+ */
+class LocalizedString extends String {
+    /**
+     * Returns formatted LocalizedString.
+     * Replaces each placeholder value (e.g. {0}, {1}) with corresponding argument.
+     * 
+     * E.g.:
+     * > new LocalizedString('{0} and {1} and {0}').format('A', 'B');
+     * A and B and A
+     *
+     * @param  {...any} args List of arguments for placeholders.
+     */
+    format(...args) {
+        let sub = this;
```

Comment:
I believe this reference is redundant if you're using arrow function in next line.

Reply:
Can you elaborate on this further?

Reply:
I mean you could just use `this` inside arrow function, it will not create a new context:
`this = this.split(`{${index}}`).join(arg);`

Reply:
`this` reference is immutable, trying to assign value for it gives `Uncaught SyntaxError: Invalid left-hand side in assignment`

Reply:
Oh, you're absolutely right, my mistake, sorry :).

However still, I'd try to avoid creating and mutating this kind of 'helper' variables and rather make use of map/reduce etc. In this case it could be something like:
```
format(...args) {
    const sub = args.reduce((string, arg, index) => string.split(`{${index}}`).join(arg), this);
    
    return new LocalizedString(sub);
}
```
I think it's more clear, but I can agree it's just a preference, so I do not expect you to do it this way. Up to you.